server/src/domain/asset/response-dto/asset-response.dto.ts

```diff
@@ -42,6 +44,7 @@ export function mapAsset(entity: AssetEntity): AssetResponseDto {
     originalPath: entity.originalPath,
     originalFileName: entity.originalFileName,
     resized: !!entity.resizePath,
+    thumbhash: entity.thumbhash?.toString('base64') || null,
```

Comment:
```suggestion
    thumbhash: entity.thumbhash?.toString('base64') ?? null,
```

Both will work, though the intention behind null coalescing is probably clearer.

Reply:
Good point

Reply:
I personally prefer || as it also handles an empty string which is also an invalid value and should be replaced with null.

Reply:
This is where JS/TS gets annoying to work with :sweat_smile:, and sort of relates to my another comment about types.

In what instance would we expect there to be a value of ""? If `entity.thumbhash` is nullish then `?.` will result in `undefined`. If `entity.thumbhash` is a zero byte array then yeah I suppose it could be become "". Though, will it _ever_ be a zero byte array? It's nice to write software defensively, but also `entity.thumbhash` should only ever be `null` or a valid string.

Reply:
The answer to whether `""` is a valid thumbhash seems to be yes-ish.

![image](https://github.com/immich-app/immich/assets/9749173/67039ad3-2785-4444-9095-78700ffcc5c7)

Reply:
Truthy vs falsy debates come up all the time in js land. Personally, I have seen errors where `item.value ?? <default value>` is used and a value of `''` ends up causing a problem. We actually had this exact issue where nestjs would generate open-api-spec with a description and summary of an empty string, which is technically a valid title. The docusaurus library used `item.title ?? item.operationId` to generate the sidebar. The empty string actually caused the title to be blank, instead of using a fallback value of operation id. In that situation the `??` was really annoying and would not have been a problem if they'd just used `||`. I don't know why an empty string would every be considered a valid title, so why make it a point to differentiate between those two cases? 

Unless you are specifically trying to only target `null` and `undefined` - specifically _not_ target `false`, 0, `''`, I don't see a reason why you would ever prefer to use `??` over `||`. I've used it when a value is a boolean or a number and I want to use 0 or false and only set a value if it is null or undefined, but when it can be a string I always use `||` since an empty string is almost always invalid.

https://github.com/immich-app/immich/pull/1069